### PR TITLE
GOBBLIN-789: Implement a FileSystem based SpecProducer.

### PR DESCRIPTION
**Pull request by sv2000**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-789

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):
Implement a FSSpecProducer class that produces JobSpecs for consumption by FsSpecConsumer. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
FsSpecProducerTest and updated existing tests in FsScheduledJobConfigurationManagerTest and IntegrationJobRestartViaSpecSuite.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by autumnust** [comment]
Can you give a little bit more context of why we need this ? Thanks.

**Comment by sv2000** [comment]
@autumnust this class implements the SpecProducer interface for producing JobSpecs which will be consumed by a corresponding SpecConsumer. The target use case is to trigger restarts of jobs running in Gobblin cluster via JobSpecs.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FsSpecProducer.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  /** Delete a {@link Spec} being executed on {@link org.apache.gobblin.runtime.api.SpecExecutor}.
+   * @param deletedSpecURI
+   * @param headers*/
+  @Override
+  public Future<?> deleteSpec(URI deletedSpecURI, Properties headers) {
+    AvroJobSpec avroJobSpec = AvroJobSpec.newBuilder().setUri(deletedSpecURI.toString())
+        .setMetadata(ImmutableMap.of(VERB_KEY, SpecExecutor.Verb.DELETE.name()))
+        .setProperties(Maps.fromProperties(headers)).build();
+    try {
+      writeAvroJobSpec(avroJobSpec);
```
Comment (autumnust) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FsSpecProducer.java]: This looks counter-intuitive to me. Why `deleteSpec` could then invoke `writeAvroJobSpec`?

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FsSpecProducer.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  /** Update a {@link Spec} being executed on {@link org.apache.gobblin.runtime.api.SpecExecutor}.
+   * @param updatedSpec*/
+  @Override
+  public Future<?> updateSpec(Spec updatedSpec) {
+    return addSpec(updatedSpec, SpecExecutor.Verb.UPDATE);
+  }
+
+  private Future<?> addSpec(Spec spec, SpecExecutor.Verb verb) {
```
Comment (autumnust) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FsSpecProducer.java]: Shall we consider change the name of this method to `acceptSpec` or others ?

**Comment by shirshanka** [comment]
Can you check why travis is unhappy?

**Comment by shirshanka** [comment]
Subject should not end with period :)

**Comment by sv2000** [comment]
@sdas triggered a build with empty commit and the build is successful.

**Pull request closed without merging**